test/createEpicMiddleware-spec.js

```diff
@@ -414,4 +418,38 @@ describe('createEpicMiddleware', () => {
 
     expect(epic.called).to.equal(true);
   });
+
+  it('should resolve once all epics will complete', async () => {
+    const reducer = (state = [], action) => state.concat(action.type);
+    const createFakeEpic = (type, delayTime = 0) => action$ => action$
+      .ofType(type)
+      .pipe(
+        ofType(type),
+        // simulate some async task
+        delay(delayTime),
+        map(({ type }) => ({ type: `${type}:completed` })),
+      );
+
+    const middleware = createEpicMiddleware();
+    const store = createStore(reducer, applyMiddleware(middleware));
+
+    middleware.run(combineEpics(
+      createFakeEpic('baz', 20),
+      createFakeEpic('bar', 10),
+      createFakeEpic('foo'),
+    ));
```

Comment:
this test would only pass as long as all the epics will return an `Observable` from `action$`...

adding something like this would cause it to fail:

```typescript
// this would not be terminated by action$
    const epicFromState = (action$, state$) => state$
      .pipe(
        filter(state => state.length),
        delay(100),
        map(payload => ({ type: 'log', payload }))
      );
```

Reply:
may `stateSubject.complete()` handle this case?